### PR TITLE
8278965: crash in SymbolTable::do_lookup

### PR DESCRIPTION
**Pull request by coleenp**
Please review this small change to hold the refcount of a super class name while calling handle_parallel_super_load. Contrary to what I wrote in JDK-8298061, the refcount in the Placeholder table should never go to one.  I can reproduce this with btree012 occasionally, so this fixes this too.

The include in placeholders.hpp is required to make the gtest compiled since placeholders header file has Symbol::maybe_increment_refcount and decrement calls.

Tested with the graal test that failed, and tier1-7.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8278965](https://bugs.openjdk.org/browse/JDK-8278965): crash in SymbolTable::do_lookup

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [3f2bea79](https://git.openjdk.org/jdk/pull/12491/files/3f2bea798ebc9c27fb443b42c9fcf23d00721d30)
 * [Frederic Parain](https://openjdk.org/census#fparain) (@fparain - Committer)
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12491/head:pull/12491` \
`$ git checkout pull/12491`

Update a local copy of the PR: \
`$ git checkout pull/12491` \
`$ git pull https://git.openjdk.org/jdk pull/12491/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12491`

View PR using the GUI difftool: \
`$ git pr show -t 12491`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12491.diff">https://git.openjdk.org/jdk/pull/12491.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/gtest/classfile/test_placeholders.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#include "runtime/mutexLocker.hpp"
+#include "threadHelper.inline.hpp"
+#include "unittest.hpp"
+
+// Test that multiple threads calling handle_parallel_super_load don't underflow supername refcount.
+TEST_VM(PlaceholderTable, supername) {
+  JavaThread* THREAD = JavaThread::current();
+  JavaThread* T2 = THREAD;
+  // the thread should be in vm to use locks
+  ThreadInVMfromNative ThreadInVMfromNative(THREAD);
```
Comment (dholmes-ora) [inline test/hotspot/gtest/classfile/test_placeholders.cpp:39]: Nit: The variable name would normally be a shorter acronym `tivmfn`.

**Review by dholmes-ora — APPROVED** [review]
Functional change seems fine. I can't say I can follow the test in detail.

Thanks.

File: test/hotspot/gtest/classfile/test_placeholders.cpp (hunk 1 above)
Comment (coleenp) [reply test/hotspot/gtest/classfile/test_placeholders.cpp:39]: That's odd that it works. I cut/pasted it from another test.

**Review by fparain — APPROVED** [review]
Looks like a reasonable fix for the issue described in the CR.

**Review by iklam — APPROVED** [review]
LGTM.

**Comment by coleenp** [comment]
Thank you Fred and Ioi!
/integrate

**Pull request closed without merging**